File: app/src/main/java/com/mongst/vinlist2/VinDao.java
package com.mongst.vinlist2;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface VinDao {

    @Query("SELECT * FROM Vin")
    List<Vin> getAllVins();

    @Insert
    void insertAll(Vin vins);

    @Update
    public abstract void updateVin(Vin... vins);

      @Delete
    void deleteVin(Vin... vins);


}

File: app/src/main/java/com/mongst/vinlist2/VinDao.kt
package com.mongst.vinlist2

import androidx.room.*
import androidx.room.Update

@Dao
interface VinDao {

    @Query("SELECT * FROM Vin")
    fun allVins(): List<Vin>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vins: Vin)

    @Update
    fun updateVin(vararg vins: Vin)

    @Delete
    fun deleteVin(vararg vins: Vin)


}
